SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp

```diff
@@ -113,3 +133,76 @@ void USpatialDispatcher::ProcessOps(Worker_OpList* OpList)
 		}
 	}
 }
+
+bool USpatialDispatcher::IsExternalSchemaOp(Worker_Op* Op) const
+{
+	Worker_ComponentId ComponentId = GetComponentId(Op);
+	return SpatialConstants::MIN_EXTERNAL_SCHEMA_ID <= ComponentId && ComponentId <= SpatialConstants::MAX_EXTERNAL_SCHEMA_ID;
+}
+
+void USpatialDispatcher::ProcessExternalSchemaOp(Worker_Op* Op)
+{
+	auto TryUserCallback = [&](Worker_ComponentId ComponentId, TFunction<void(UOpCallbackTemplate*)>& OpCallback)
+	{
+		if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(ComponentId))
+		{
+			UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+			OpCallback(UserCallback);
+		}
+	};
+	Worker_ComponentId ComponentId = GetComponentId(Op);
+	check(ComponentId != SpatialConstants::INVALID_COMPONENT_ID);
+
+	UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(ComponentId);
+	if (UserCallbackWrapper == nullptr)
+	{
+		return;
+	}
+	UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+	
+	switch (Op->op_type)
+	{
+	case WORKER_OP_TYPE_ADD_COMPONENT:
+		UserCallback->OnAddComponent(Op->add_component);
+		break;
+	case WORKER_OP_TYPE_REMOVE_COMPONENT:
+		UserCallback->OnRemoveComponent(Op->remove_component);
+		break;
+	case WORKER_OP_TYPE_COMPONENT_UPDATE:
+		UserCallback->OnComponentUpdate(Op->component_update);
+		break;
+	case WORKER_OP_TYPE_AUTHORITY_CHANGE:
+		UserCallback->OnAuthorityChange(Op->authority_change);
+		break;
+	case WORKER_OP_TYPE_COMMAND_REQUEST:
+		UserCallback->OnCommandRequest(Op->command_request);
+		break;
+	case WORKER_OP_TYPE_COMMAND_RESPONSE:
+		UserCallback->OnCommandResponse(Op->command_response);
+		break;
+	default:
+		checkNoEntry();
```

Comment:
Add comment to indicate that we should only get here if `GetComponentId` returns a valid ID for a new op type and this function hasn't been updated.